src/toolkits/drawing_classifier/drawing_classifier.cpp

```diff
@@ -349,9 +349,24 @@ void drawing_classifier::init_training(
   std::vector<std::string> gpu_names = training_compute_context_->gpu_names();
   print_training_device(gpu_names);
 
-  // Initialize the neural net. Note that this depends on statistics computed
-  // by the data iterator.
-  nn_spec_ = init_model();
+  // If warm_start enabled, an 'mlmodel_path' is passed in options.
+  // Else, call init_model() to initialize the neural net using xavier.
+  if (opts.find("mlmodel_path") != opts.end() ) {
+    // Initialize the neural net with warm start model weights.
+    auto mlmodel_path_iter = opts.find("mlmodel_path");
+    if (mlmodel_path_iter == opts.end()) {
+      log_and_throw("Expected option \"mlmodel_path\" not found.");
+    }
+    const std::string mlmodel_path = mlmodel_path_iter->second;
```

Comment:
no need to const. just use `const &`. Don't erase it from the option. It's good for debugging.

Reply:
I mean it's better to keep `opts` intact since it's passed from the user layer.

Reply:
In this particular case, we should delete it because we don't want to surface it in the model state. This isn't something the user can specify manually for now

Using `const&` to avoid a copy would still be nice, though.

Reply:
I believe we don't surface that in our metadata section unless we explicitly want to expose that. Correct me if I'm wrong.

Reply:
Everything we put into the `state` object will be exposed to the user in Python, via this mechanism: https://github.com/apple/turicreate/blob/b5e16c58d75913ad2a2dd4d4a05b631145fdff4c/src/model_server/lib/extensions/ml_model.cpp#L24

Reply:
glad to know this path! I didn't notice this hidden part.

Reply:
if @fareeha wants to use the reference of the string, make sure you defer to delete the iterator after you use the string reference variable. Or you can move itr->second to your string and then delete it immediately.